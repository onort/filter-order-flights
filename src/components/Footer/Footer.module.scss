@import "../../styles/main.scss";

.footerContainer {
  display: grid;
  width: 1120px;
  padding: map-get($spacers, x-large) 0;
  margin: map-get($spacers, x-large) auto;
  grid-column-gap: px-to-rem(50px);
  grid-template-columns: 3fr 7fr;
}

.textContent {
  color: $grey2;
  font-family: $font-family-primary;
  font-size: map-get($font-sizes, small);
  font-weight: 400;
  line-height: px-to-rem(24px);
  text-decoration: none;
}

.menu {
  display: grid;
  flex: 5;
  grid-template-columns: repeat(auto-fit, minmax(160px, 1fr));
}
