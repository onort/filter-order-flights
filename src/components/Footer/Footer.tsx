import React from "react"

import styles from "./Footer.module.scss"
import { Container, FooterMenuGroup } from "../"
import { ThyLogo } from "../../assets/images"

export const menuGroupA = {
  links: [
    { text: "Anasayfa", uri: "#" },
    { text: "Şehir & Gezi Rehberi", uri: "#" },
    { text: "Fırsatlar", uri: "#" },
    { text: "Keşfet", uri: "#" },
    { text: "Yardım", uri: "#" },
    { text: "İletişim", uri: "#" }
  ],
  title: "Sayfalar"
}

export const menuGroupB = {
  links: [
    { text: "İstanbul", uri: "#" },
    { text: "Ankara", uri: "#" },
    { text: "İzmir", uri: "#" },
    { text: "New York", uri: "#" },
    { text: "Roma", uri: "#" },
    { text: "Venedik", uri: "#" },
    { text: "Amsterdam", uri: "#" }
  ],
  title: "Popüler Şehirler"
}

export const menuGroupC = {
  links: [
    { text: "Online Check-in", uri: "#" },
    { text: "Genel Uçuş Kuralları", uri: "#" },
    { text: "Yolcu Hakları", uri: "#" }
  ],
  title: "Merak Edilenler"
}

export const menuGroupD = {
  links: [
    { text: "İletişim", uri: "#" },
    { text: "Twitter", uri: "#" },
    { text: "Facebook", uri: "#" }
  ],
  title: "Bize Ulaşın"
}

const Footer: React.SFC = () => {
  return (
    <footer className={styles.footer}>
      <Container className={styles.footerContainer}>
        <div className={styles.branding}>
          <img className={styles.logo} src={ThyLogo} alt="Logo" />
          <p className={styles.textContent}>
            Fusce vehicula dolor arcu, sit amet blandit dolor mollis nec. Donec
            viverra eleifend lacus, vitae ullamcorper metus. Sed sollicitudin
            ipsum quis nunc sollicitudin ultrices.
          </p>
        </div>
        <div className={styles.menu}>
          <FooterMenuGroup title={menuGroupA.title} links={menuGroupA.links} />
          <FooterMenuGroup title={menuGroupB.title} links={menuGroupB.links} />
          <FooterMenuGroup title={menuGroupC.title} links={menuGroupC.links} />
          <FooterMenuGroup title={menuGroupD.title} links={menuGroupD.links} />
        </div>
      </Container>
    </footer>
  )
}

export default Footer
