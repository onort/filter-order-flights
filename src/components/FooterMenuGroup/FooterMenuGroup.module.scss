@import "../../styles/main.scss";

.title {
  color: $grey1;
  font-family: $font-family-primary;
  font-size: map-get($heading-sizes, h5);
  font-weight: 600;
}

.list {
  padding: 0;
  list-style: none;
}

.item {
  a {
    color: $grey2;
    font-family: $font-family-primary;
    font-size: map-get($font-sizes, small);
    font-weight: 400;
    line-height: px-to-rem(24px);
    text-decoration: none;
  }
}
