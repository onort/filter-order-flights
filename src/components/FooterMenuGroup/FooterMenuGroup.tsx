import React from "react"

import styles from "./FooterMenuGroup.module.scss"

interface Link {
  text: string
  uri: string
}

interface Props {
  links: Link[]
  title: string
}

const FooterMenuGroup: React.SFC<Props> = props => {
  const { links, title } = props
  return (
    <div className={styles.container}>
      <h5 className={styles.title}>{title}</h5>
      <ul className={styles.list}>
        {links.map(link => (
          <li key={link.text} className={styles.item}>
            <a href={link.uri}>{link.text}</a>
          </li>
        ))}
      </ul>
    </div>
  )
}

export default FooterMenuGroup
