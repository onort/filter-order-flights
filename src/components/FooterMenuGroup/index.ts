import FooterMenuGroup from "./FooterMenuGroup"

export default FooterMenuGroup
