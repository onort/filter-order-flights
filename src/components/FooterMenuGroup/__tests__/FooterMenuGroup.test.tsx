import React from "react"
import { shallow } from "enzyme"

import FooterMenuGroup from "../"
import { menuGroupD } from "../../Footer/Footer"

describe("<FooterMenuGroup />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(
      <FooterMenuGroup title={menuGroupD.title} links={menuGroupD.links} />
    )
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should have correct title", () => {
    const wrapper = shallow(
      <FooterMenuGroup title={menuGroupD.title} links={menuGroupD.links} />
    )
    expect(wrapper.find("h5").text()).toBe(menuGroupD.title)
  })

  it("should have correct amount of links", () => {
    const wrapper = shallow(
      <FooterMenuGroup title={menuGroupD.title} links={menuGroupD.links} />
    )
    expect(wrapper.find("a")).toHaveLength(menuGroupD.links.length)
  })
})
