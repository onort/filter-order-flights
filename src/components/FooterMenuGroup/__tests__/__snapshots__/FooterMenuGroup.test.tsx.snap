// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FooterMenuGroup /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FooterMenuGroup
    links={
      Array [
        Object {
          "text": "İletişim",
          "uri": "#",
        },
        Object {
          "text": "Twitter",
          "uri": "#",
        },
        Object {
          "text": "Facebook",
          "uri": "#",
        },
      ]
    }
    title="Bize Ulaşın"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h5
          className="title"
        >
          Bize Ulaşın
        </h5>,
        <ul
          className="list"
        >
          <li
            className="item"
          >
            <a
              href="#"
            >
              İletişim
            </a>
          </li>
          <li
            className="item"
          >
            <a
              href="#"
            >
              Twitter
            </a>
          </li>
          <li
            className="item"
          >
            <a
              href="#"
            >
              Facebook
            </a>
          </li>
        </ul>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Bize Ulaşın",
          "className": "title",
        },
        "ref": null,
        "rendered": "Bize Ulaşın",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="item"
            >
              <a
                href="#"
              >
                İletişim
              </a>
            </li>,
            <li
              className="item"
            >
              <a
                href="#"
              >
                Twitter
              </a>
            </li>,
            <li
              className="item"
            >
              <a
                href="#"
              >
                Facebook
              </a>
            </li>,
          ],
          "className": "list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "İletişim",
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="#"
              >
                İletişim
              </a>,
              "className": "item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "İletişim",
                "href": "#",
              },
              "ref": null,
              "rendered": "İletişim",
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "Twitter",
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="#"
              >
                Twitter
              </a>,
              "className": "item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Twitter",
                "href": "#",
              },
              "ref": null,
              "rendered": "Twitter",
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "Facebook",
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="#"
              >
                Facebook
              </a>,
              "className": "item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Facebook",
                "href": "#",
              },
              "ref": null,
              "rendered": "Facebook",
              "type": "a",
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5
            className="title"
          >
            Bize Ulaşın
          </h5>,
          <ul
            className="list"
          >
            <li
              className="item"
            >
              <a
                href="#"
              >
                İletişim
              </a>
            </li>
            <li
              className="item"
            >
              <a
                href="#"
              >
                Twitter
              </a>
            </li>
            <li
              className="item"
            >
              <a
                href="#"
              >
                Facebook
              </a>
            </li>
          </ul>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Bize Ulaşın",
            "className": "title",
          },
          "ref": null,
          "rendered": "Bize Ulaşın",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="item"
              >
                <a
                  href="#"
                >
                  İletişim
                </a>
              </li>,
              <li
                className="item"
              >
                <a
                  href="#"
                >
                  Twitter
                </a>
              </li>,
              <li
                className="item"
              >
                <a
                  href="#"
                >
                  Facebook
                </a>
              </li>,
            ],
            "className": "list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "İletişim",
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="#"
                >
                  İletişim
                </a>,
                "className": "item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "İletişim",
                  "href": "#",
                },
                "ref": null,
                "rendered": "İletişim",
                "type": "a",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "Twitter",
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="#"
                >
                  Twitter
                </a>,
                "className": "item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Twitter",
                  "href": "#",
                },
                "ref": null,
                "rendered": "Twitter",
                "type": "a",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "Facebook",
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="#"
                >
                  Facebook
                </a>,
                "className": "item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Facebook",
                  "href": "#",
                },
                "ref": null,
                "rendered": "Facebook",
                "type": "a",
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<FooterMenuGroup /> should match snapshot 2`] = `"<div class=\\"container\\"><h5 class=\\"title\\">Bize Ulaşın</h5><ul class=\\"list\\"><li class=\\"item\\"><a href=\\"#\\">İletişim</a></li><li class=\\"item\\"><a href=\\"#\\">Twitter</a></li><li class=\\"item\\"><a href=\\"#\\">Facebook</a></li></ul></div>"`;
